source/slang/slang-emit-spirv.cpp

```diff
@@ -1415,12 +1415,24 @@ struct SPIRVEmitContext
                     if (m_decoratedSpvInsts.add(getID(resultSpvType)))
                     {
                         IRSizeAndAlignment sizeAndAlignment;
-                        getNaturalSizeAndAlignment(m_targetProgram->getOptionSet(), ptrType->getValueType(), &sizeAndAlignment);
+                        uint32_t stride;
```

Comment:
Note that this should not only apply to arrays, but also struct types that contains unsized arrays (and therefore become unsized).

instead of special casing unsized arrays, you can simply call getNatrualSizeAndAlignment on the pointee type like the code is doing, and if stride is 0 then emit 0xFFFF.